crates/outbound-http/src/host_impl.rs

```diff
@@ -55,15 +56,20 @@ impl outbound_http::Host for OutboundHttp {
 
             let method = method_from(req.method);
 
-            let abs_url = if req.uri.starts_with('/') {
-                format!("{}{}", self.origin, req.uri)
+            let mut headers = request_headers(req.headers).map_err(|_| HttpError::RuntimeError)?;
+            let abs_url;
+            if req.uri.starts_with('/') {
+                abs_url = format!("{}{}", self.origin, req.uri);
+                // If it is a self request - set the host header.
+                let uri = Uri::from_str(&req.uri).map_err(|_| HttpError::RuntimeError)?;
+                let host = format!("{}:{}", uri.host().unwrap(), uri.port().unwrap());
```

Comment:
I'm a bit puzzled as to how this works on what seems to be the original request URI, the one that starts with `/` - I don't quite get how that would provide you with a host.  On the other hand if it works it works...!

Reply:
I did not test the older outbound-http so I am not sure there, Will write a test for it using the JS SDK.